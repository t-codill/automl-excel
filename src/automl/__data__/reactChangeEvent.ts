// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
