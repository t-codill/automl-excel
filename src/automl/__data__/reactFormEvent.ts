// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
