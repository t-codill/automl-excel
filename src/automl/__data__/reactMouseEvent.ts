// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
