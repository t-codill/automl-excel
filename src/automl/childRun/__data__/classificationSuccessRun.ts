import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const classificationSuccessRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_001",
        parentRunId: "AutoML_000",
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: {
            iteration: "9",
            run_algorithm: "Ensemble",
            run_preprocessor: "rpp1"
        }
    },
    runMetrics: {
        scalarMet1: "1",
        scalarMet2: "2",
        scalarMet3: "3",
        accuracy_table: {},
        confusion_matrix: {}
    },
    modelUri: "sampleModelUri",
    experimentName: "classificationSuccess"
};
