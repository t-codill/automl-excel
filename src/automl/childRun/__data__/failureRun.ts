import { NotNullableProperties } from "../../common/utils/NotNullableProperties";
import { IChildRunData } from "../ChildRun";

export const failureRun: NotNullableProperties<IChildRunData> = {
    run: {
        runId: "AutoML_002",
        status: "Failed",
        parentRunId: "AutoML_000",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:20:30.000Z"),
        properties: undefined
    },
    runMetrics: {},
    experimentName: ""
};
