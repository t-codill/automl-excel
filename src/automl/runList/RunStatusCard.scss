@import "../Themes.scss";

.run-status-card {

    @include primaryTextColor;
    font-size: 20px;
    border: 1px solid #0392cf;
    padding: 1em .5em;
    margin: 0 .5em 1em 0;
    box-sizing: border-box;
    text-align: center;
    cursor: pointer;

    .filter-tag {
        text-decoration: underline;
        color: #0392cf;
        font-size: 13px;
    }

    &:hover {

        @include runStatusCardHoverBackgroundColor;
    }

    &.active-card {
        color: #ffffff;
        background-color: #015cda;

        &:hover {
            background-color: #015cda;
        }

        .filter-tag {
            color: #ffffff;
        }
    }
}