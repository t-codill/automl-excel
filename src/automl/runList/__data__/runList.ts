import { IRunDtoWithExperimentName } from "../../services/RunHistoryService";

export const runList: IRunDtoWithExperimentName[] | undefined = [
    {
        runNumber: 1,
        runId: "AutoML_001",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2019-02-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-02-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: `{"primary_metric": "AUC_weighted",
          "data_script": null,\
          "compute_target": "local",
          "n_cross_validations": 3,
          "max_cores_per_iteration": 1,
          "blacklist_algos": [],
          "exit_score": 100,
          "name": "my_experiment",
          "task_type": "classification" }`
        },
        tags: {
            iterations: "2"
        },
        experimentName: "AAA"
    },
    {
        runNumber: 2,
        runId: "AutoML_002",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2019-01-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-01-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: `{"primary_metric": "AUC_weighted",
          "data_script": null,\
          "compute_target": "local",
          "n_cross_validations": 3,
          "max_cores_per_iteration": 1,
          "blacklist_algos": [],
          "exit_score": 100,
          "name": "my_experiment",
          "task_type": "classification" }`
        },
        tags: {
            iterations: "2"
        },
        experimentName: "AAA"
    },
    {
        runNumber: 3,
        runId: "AutoML_003",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2018-01-01T08:20:00.000Z"),
        endTimeUtc: new Date("2018-01-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: `{"primary_metric": "AUC_weighted",
          "data_script": null,\
          "compute_target": "local",
          "n_cross_validations": 3,
          "max_cores_per_iteration": 1,
          "blacklist_algos": [],
          "exit_score": 100,
          "name": "my_experiment",
          "task_type": "classification" }`
        },
        tags: {
            iterations: "2"
        },
        experimentName: "BBB"
    },
    {
        runNumber: 4,
        runId: "AutoML_004",
        parentRunId: undefined,
        status: "Failed",
        startTimeUtc: new Date("2019-04-01T08:20:00.000Z"),
        endTimeUtc: new Date("2019-04-01T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: `{"primary_metric": "AUC_weighted",
          "data_script": null,\
          "compute_target": "local",
          "n_cross_validations": 3,
          "max_cores_per_iteration": 1,
          "blacklist_algos": [],
          "exit_score": 100,
          "name": "my_experiment",
          "task_type": "classification" }`
        },
        tags: {
            iterations: "2"
        },
        experimentName: "BBB"
    },
    {
        runNumber: 5,
        runId: "AutoML_005",
        parentRunId: undefined,
        status: "Completed",
        startTimeUtc: new Date("2018-01-11T08:20:00.000Z"),
        endTimeUtc: new Date("2018-01-11T08:30:00.000Z"),
        properties: {
            AMLSettingsJsonString: `{"primary_metric": "AUC_weighted",
          "data_script": null,\
          "compute_target": "local",
          "n_cross_validations": 3,
          "max_cores_per_iteration": 1,
          "blacklist_algos": [],
          "exit_score": 100,
          "name": "my_experiment",
          "task_type": "classification" }`
        },
        tags: {
            iterations: "2"
        },
        experimentName: "BBB"
    }
];
