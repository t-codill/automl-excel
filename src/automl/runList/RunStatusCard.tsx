import { FocusZone } from "office-ui-fabric-react";
import * as React from "react";

import "./RunStatusCard.scss";

export interface IRunStatusCardProps {
    runStatusLabel: string;
    runStatusCount: number;
    isActive: boolean;
    className: string;
    handleStatusFilterChange(
        statusFilter: string | undefined
    ): void;
}
export class RunStatusCard extends React.Component<IRunStatusCardProps>{
    public readonly render = (): React.ReactNode => {
        let filterText = "Filter";
        let className = "run-status-card";
        if (this.props.isActive) {
            className += " active-card";
            filterText = "Remove Filter";
        }
        return (
            <FocusZone key={this.props.runStatusLabel}
                className={this.props.className}
                onClick={this.statusChange}>
                <div className={className}>
                    <div>{this.props && this.props.runStatusLabel}</div>
                    <div>{this.props && this.props.runStatusCount}</div>
                    <div className="filter-tag">{filterText}</div>
                </div>
            </FocusZone >
        );
    }
    public statusChange = (_event: React.MouseEvent<HTMLElement>): void => {
        if (this.props.isActive) {
            this.props.handleStatusFilterChange(undefined);
        }
        else {
            this.props.handleStatusFilterChange(this.props.runStatusLabel);
        }
    }
}
