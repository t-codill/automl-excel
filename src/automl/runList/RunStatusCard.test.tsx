import { shallow, ShallowWrapper } from "enzyme";
import { FocusZone } from "office-ui-fabric-react";
import * as React from "react";
import { reactMouseEvent } from "../__data__/reactMouseEvent";
import { RunStatusCard } from "./RunStatusCard";

describe("RunStatusCard", () => {
    let tree: ShallowWrapper;
    const mockCallBack = jest.fn();
    describe("Active Card", () => {
        beforeEach(() => {
            const props = {
                runStatusLabel: "Failed",
                runStatusCount: 123,
                isActive: true,
                handleStatusFilterChange: mockCallBack,
                className: ""
            };
            tree = shallow(
                <RunStatusCard {...props} />
            );
            mockCallBack.mockClear();
        });
        it("should render", () => {
            expect(tree)
                .toMatchSnapshot();
        });
        it("should deactivate filter", () => {
            const runStatusCardProps = tree.find(FocusZone)
                .props();
            if (runStatusCardProps && runStatusCardProps.onClick) {
                runStatusCardProps.onClick(reactMouseEvent);
            }
            expect(mockCallBack)
                .toBeCalledWith(undefined);
        });
    });
    describe("Inactive Card", () => {
        beforeEach(() => {
            const props = {
                runStatusLabel: "Failed",
                runStatusCount: 123,
                isActive: false,
                handleStatusFilterChange: mockCallBack,
                className: ""
            };
            tree = shallow(
                <RunStatusCard {...props} />
            );
            mockCallBack.mockClear();
        });
        it("should render", () => {
            expect(tree)
                .toMatchSnapshot();
        });
        it("should active filter", () => {
            const runStatusCardProps = tree.find(FocusZone)
                .props();
            if (runStatusCardProps && runStatusCardProps.onClick) {
                runStatusCardProps.onClick(reactMouseEvent);
            }
            expect(mockCallBack)
                .toBeCalledWith("Failed");
        });
    });
});
