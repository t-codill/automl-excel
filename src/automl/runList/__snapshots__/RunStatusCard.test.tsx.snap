// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunStatusCard Active Card should render 1`] = `
<FocusZone
  className=""
  direction={2}
  isCircularNavigation={false}
  key="Failed"
  onClick={[Function]}
>
  <div
    className="run-status-card active-card"
  >
    <div>
      Failed
    </div>
    <div>
      123
    </div>
    <div
      className="filter-tag"
    >
      Remove Filter
    </div>
  </div>
</FocusZone>
`;

exports[`RunStatusCard Inactive Card should render 1`] = `
<FocusZone
  className=""
  direction={2}
  isCircularNavigation={false}
  key="Failed"
  onClick={[Function]}
>
  <div
    className="run-status-card"
  >
    <div>
      Failed
    </div>
    <div>
      123
    </div>
    <div
      className="filter-tag"
    >
      Filter
    </div>
  </div>
</FocusZone>
`;
